scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2274,6 +2269,21 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
     }
   }
 
+  def condExprInParens[T <: Token: TokenInfo]: Term = {
+    if (dialect.allowSignificantIndentation) {
```

Comment:
i think this branch is not tested. you don't use scala 2 in any test cases?

Reply:
For scala2 this part branch isn't involved because the condition has to be inside braces.
There are tests for scala2 `if`/`while` in [TermSuite](https://github.com/scalameta/scalameta/blob/main/tests/shared/src/test/scala/scala/meta/tests/parsers/TermSuite.scala).